components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java

```diff
@@ -316,5 +319,51 @@ public void updateRoleName(int tenantId, String oldRoleName, String newRoleName)
             throw new IdentitySCIMException("Error when updating role name of the role: " + oldRoleName);
         }
     }
+    /**
+     * Lists the Groups created from SCIM with a attribute filter and search regex
+     *
+     * @return list of SCIM groups
+     * @throws IdentitySCIMException
+     */
+    public String[] getGroupNameList(String searchAttributeName, String searchAttributeValue, Integer tenantId)
+            throws IdentitySCIMException {
+
+        List<String> roleList = new ArrayList<>();
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
```

Comment:
shall we use try-with resources for this? It should be similar to [1]
[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java#L97